### PR TITLE
Added fetch metadata Policy

### LINKED ISSUE #5: Create an interface for Fetch-Metadata Policies
## Expected Feature
Construct an interface for Fetch-Metadata policies. This is to provide a default "first party" policy, while also allowing developers with more specific needs to construct their own policies based on an interface provided by the bundle. This will also allow us to build our own policy presets more easily and use them in the configuration tree. 

## Actual Behavior
At the moment the Fetch-Metadata policies in the bundle are controlled by the config tree. This will become unweildy quickly to be sure.

### PR DESCRIPTION
**Pull request by henrym2**
Added Policy interface for fetch metadata
Added Default Policy
Modified Request Subscriber to use new policy interface
Created new tests for Fetch Metadata policy and Subscriber
Updated Config to include new configuration
Updated service configuration to include new fetch metadata policy service
Updated configuration to allow for custom policy to be assigned
Updated config to include new endpoint exemptions

Fixes #5 
> It's a good idea to open an issue first for discussion.

- [x] Tests pass
- [x] Appropriate changes to README are included in PR

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -14,40 +15,48 @@ public function getConfigTreeBuilder()
...
                     ->children()
                         ->booleanNode('active')->defaultFalse()->end()
                     ->end()
                 ->end()
-                ->arrayNode('fetch_metadata')
-                    ->children()
-                        ->booleanNode('active')->defaultFalse()->end()
-                    ->end()
-                ->end()
+                ->append($this->getFetchMetaData())
```
Comment (rowan-m) [inline DependencyInjection/Configuration.php]: `s/etaData/etadata/`

File: Policies/FetchMetadataDefaultPolicy.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+     * @var array
+     */
+    private $corsEndpoints;
+
+    public function __construct($corsEndpoints = [])
+    {
+        $this->corsEndpoints = $corsEndpoints;
+    }
+    /**
+     * apllyPolicy Applies the default Fetch Metadata Policy as defined by https://web.dev/fetch-metadata/
```
Comment (rowan-m) [inline Policies/FetchMetadataDefaultPolicy.php]: `s/aplly/apply/`

**Review by rowan-m — CHANGES_REQUESTED** [review]
Just scanned through and picked out minor typos. Will keep going.

File: Policies/FetchMetadataDefaultPolicy.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+use Symfony\Component\HttpFoundation\Request;
+
+class FetchMetadataDefaultPolicy implements FetchMetadataPolicyInterface
+{
+    /**
+     * CorsEndpoints denotes the allowed cross origin endpoints as part of the Default Fetch metadata policy
+     *
+     * @var array
+     */
+    private $corsEndpoints;
```
Comment (rowan-m) [inline Policies/FetchMetadataDefaultPolicy.php:15]: What's the format for this? Array of URL string?

**Review by rowan-m — APPROVED** [review]
Happy to get this merged so we can test it out and iterate. I have a feeling that the policy may want some logging options and possibly a more nuanced return type or option for hooking in different behaviour.

File: Policies/FetchMetadataDefaultPolicy.php (hunk 3 above)
Comment (henrym2) [reply Policies/FetchMetadataDefaultPolicy.php:15]: Yes, I tried to set it to a more...reasonable identifier like a string array. But I ended up getting a number of angry responses from Symfony's auto-wire service. It's something I'll continue to iterate with. Until then though I'll update that with an example for the input, and then merge this into main.

**Pull request merged**